bin/validate_sample_meta.pl

```diff
@@ -0,0 +1,183 @@
+#! /usr/bin/perl
+
+# perl-5.16.3 -I /software/CGP/canpipe/live/lib/perl5 validate_smaple_meta.pl
+use strict;
+use warnings FATAL => 'all';
+use autodie qw(:all);
+use Getopt::Long;
+use Pod::Usage qw(pod2usage);
+use Const::Fast qw(const);
+use Carp;
+
+use Spreadsheet::Read qw(ReadData);
+use Spreadsheet::WriteExcel; # to write xls output
+use Excel::Writer::XLSX; # to write xlsx output
+
+use Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader;
+
+my $opts = setup_options();
+my $input = $opts->{'i'};
+my $output = $opts->{'o'};
+my $check_all = 'n';
+if (defined $opts->{'a'}) {
+  $check_all = 'y';
+}
+#print "check_all setting: $check_all.\n";
+my $format;
+
+if (defined $opts->{'f'} && $opts->{'f'} =~ m/^xls$|^xlsx$|^tsv$/i) {
+  $format = $opts->{'f'};
+} elsif (! defined $opts->{'f'}) {
+  $format = 'tsv';
+} else {
+  die("--------\nError: Does not recognise the specified file format, accept values are: 'tsv', 'xls' and 'xlsx'.\n--------\n");
+}
+
+my @in;
+### read in input as array of lines (concatenated columns with tabs)
+if ($format eq 'xlsx'|| $format eq 'xls') {
+  my $book = ReadData ($input);
+  my @rows = Spreadsheet::Read::rows($book->[1]);
+  foreach my $row (@rows) {
+    foreach my $cell (@$row) {
+      $cell = '' if (! defined $cell);
+      # print $cell."\n";
+    }
+    push @in, (join "\t", @$row);
+  }
+} else { # assume it's a tsv
+  open (my $FH, '<', $input) or croak("--------\nError trying to open input file $input: $!\n--------\n");
+  chomp(@in = <$FH>);
+  close $FH;
+}
+
+my ($valid_results, $valid_status) = Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader::validate_samples(\@in, $check_all);
```

Comment:
This should be an object class with a new function taking the path to the spreadsheet.

e.g.

```perl
my $obj = Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader->new($input, $output);
if($obj->validate) {
  $obj->write_validated(); # hide the tsv/xls/xlsx here
}

```

The new function then can set state information via an `_init()` call.  During the function it can set the input/output format in to an object variable which can be referenced in other object calls.  See [here](https://github.com/ICGC-TCGA-PanCancer/PCAP-core/blob/c60cf099a591392f4e2ea85f47e036a57b4890b5/lib/PCAP/Bam/Coverage.pm#L19-L34) for a relatively simple example.